### PR TITLE
Feat: Structurelize patch object

### PR DESCRIPTION
**Pull request by yue9944882**
@brendandburns we saw that many developers are trying to pass an object directly into the patch call, making this structurelized can prevent them from that. additionally, this pull moves the patch format consts into the custom.V1Patch, which makes it easier to find

### CODE REVIEW

**Comment by brendandburns** [comment]
Did you change the code generator to do this? We need to make sure we can re-generate the code going forward...

**Comment by yue9944882** [comment]
@brendandburns yep the change in generator is submitted here. https://github.com/kubernetes-client/gen/pull/118. it's basically adding a type mapping to the xml configuration, i noticed the python client has already done the mapping for `V1Patch` class.

**Comment by yue9944882** [comment]
https://github.com/kubernetes/kubernetes/blob/3d4124f2e083591c98e8a874760011781d3ee15c/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L737

the endpoint for patch api endpoint is defined w/ structurelized schema but the pre-process script in the generator somehow omitted that.

File: kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.io.IOException;
+
+@JsonAdapter(V1Patch.V1PatchAdapter.class)
+public class V1Patch {
+
+  public static final String PATCH_FORMAT_JSON_PATCH = "application/json-patch+json";
+  public static final String PATCH_FORMAT_JSON_MERGE_PATCH = "application/merge-patch+json";
+  public static final String PATCH_FORMAT_STRATEGIC_MERGE_PATCH = "application/strategic-merge-patch+json";
+
+  public V1Patch() {
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java]: Delete empty constructor?

File: kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    public void write(JsonWriter jsonWriter, V1Patch patch) throws IOException {
+      jsonWriter.jsonValue(patch.getValue());
+    }
+
+    @Override
+    public V1Patch read(JsonReader jsonReader) throws IOException {
+      throw new IOException("deserialize patch content is not supported");
+    }
+  }
+}
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java]: missing newline.

File: kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  public class V1PatchAdapter extends TypeAdapter<V1Patch> {
+    @Override
+    public void write(JsonWriter jsonWriter, V1Patch patch) throws IOException {
+      jsonWriter.jsonValue(patch.getValue());
+    }
+
+    @Override
+    public V1Patch read(JsonReader jsonReader) throws IOException {
+      throw new IOException("deserialize patch content is not supported");
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java]: use `java.lang.UnsupportedOperationException` ?

**Comment by brendandburns** [comment]
Cool. A couple of small comments and then I think this is good to go.

File: kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java (hunk 1 above)
Comment (yue9944882) [reply kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java]: done

File: kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java (hunk 3 above)
Comment (yue9944882) [reply kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java]: updated

**Comment by yue9944882** [comment]
lets get this in!

File: kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  public static final String PATCH_FORMAT_JSON_PATCH = "application/json-patch+json";
+  public static final String PATCH_FORMAT_JSON_MERGE_PATCH = "application/merge-patch+json";
+  public static final String PATCH_FORMAT_STRATEGIC_MERGE_PATCH = "application/strategic-merge-patch+json";
+
+  public V1Patch(final String value) {
+    this.value = value;
+  }
+
+  private String value;
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java]: nit: move this above the constructor.

File: kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    public void write(JsonWriter jsonWriter, V1Patch patch) throws IOException {
+      jsonWriter.jsonValue(patch.getValue());
+    }
+
+    @Override
+    public V1Patch read(JsonReader jsonReader) throws IOException {
+      throw new UnsupportedOperationException("deserializing patch data is not supported");
+    }
+  }
+
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/custom/V1Patch.java]: nit: delete empty line.

**Comment by brendandburns** [comment]
Sorry, a couple of additional nits... I'm going to lgtm, but can you address in a follow-up PR?

Thanks!

/lgtm
/approve

**Comment by runzexia** [comment]
Hi Do you have a plan to have a new client version? Need this feature very much

**Pull request merged**